WordPressKit/RemoteVideoPressVideo.swift

```diff
@@ -0,0 +1,128 @@
+    public var privacySetting: VideoPressPrivacySetting = .siteDefault
+    public var posterURL: URL?
+    public var originalURL: URL?
+    public var watermarkURL: URL?
+    public var bgColor: String?
+    public var blogId: Int?
+    public var postId: Int?
+    public var finished: Bool?
+
+    public var token: String?
+
+    enum CodingKeys: String, CodingKey {
+        case id, title, description, width, height, duration, displayEmbed, allowDownload, rating, privacySetting, posterURL, originalURL, watermarkURL, bgColor, blogId, postId, finished, token
+    }
+
+    public init(dictionary metadataDict: NSDictionary, id: String) {
+        self.id = id
+
+        title = metadataDict.string(forKey: "title")
+        videoDescription = metadataDict.string(forKey: "description")
+        width = metadataDict.number(forKey: "width")?.intValue
+        height = metadataDict.number(forKey: "height")?.intValue
+        duration = metadataDict.number(forKey: "duration")?.intValue
+        displayEmbed = metadataDict.object(forKey: "display_embed") as? Bool
+        allowDownload = metadataDict.object(forKey: "allow_download") as? Bool
+        rating = metadataDict.string(forKey: "rating")
+        if let privacySettingValue = metadataDict.number(forKey: "privacy_setting")?.intValue, let privacySettingEnum = VideoPressPrivacySetting.init(rawValue: privacySettingValue) {
+            privacySetting = privacySettingEnum
+        }
+        if let poster = metadataDict.string(forKey: "poster") {
+            posterURL = URL(string: poster)
+        }
+        if let original = metadataDict.string(forKey: "original") {
+            originalURL = URL(string: original)
+        }
+        if let watermark = metadataDict.string(forKey: "watermark") {
+            watermarkURL = URL(string: watermark)
+        }
+        bgColor = metadataDict.string(forKey: "bg_color")
+        blogId = metadataDict.number(forKey: "blog_id")?.intValue
+        postId = metadataDict.number(forKey: "post_id")?.intValue
+        finished = metadataDict.object(forKey: "finished") as? Bool
+    }
+
+    /// Returns the specified URL adding the token as a query parameter, which is required to play private videos.
+    /// - Parameters:
+    ///   - url: URL to include the token.
+    ///
+    /// - Returns: The specified URL with the token as a query parameter.
+    @objc(getURLWithToken:)
+    public func getURLWithToken(url: URL?) -> URL? {
+        guard var urlWithToken = url else {
+            return nil
+        }
+        if let token = self.token, var urlComponents = URLComponents(url: urlWithToken, resolvingAgainstBaseURL: true) {
+            let metadataTokenParam = URLQueryItem(name: "metadata_token", value: token)
+            var queryItems: [URLQueryItem] = urlComponents.queryItems ?? []
+            queryItems.append(metadataTokenParam)
+            urlComponents.queryItems = queryItems
+            urlWithToken = urlComponents.url!
```

Comment:
See comment above on a proposal that could protect against the force unwrap. If we don't go with the proposal, we should consider returning `urlWithToken` if `urlComponents.url` is `nil` (even though we know this should be safe).

Reply:
I finally applied the above proposal so the force unwrap is no longer done.